plutus-ledger-api/test-plugin/Spec/Data/Budget/9.6/currencySymbolValueOf.eval.golden

```diff
@@ -0,0 +1 @@
+(con integer 3636)
```

Comment:
Same as non-data version :heavy_check_mark:

Reply:
I suggest you to rewrite this test not to use golden file but to directly compare with constant, for example:

```haskell
testRatioInterop :: TestTree
testRatioInterop = testCase "ratioInterop" do
  runExceptT (runUPlc [getPlcNoAnn roundPlc, snd (Lift.liftProgramDef (Ratio.fromGHC 3.75))])
    >>= \case
      Left e -> assertFailure (show e)
      Right r -> r @?= Core.mkConstant () (4 :: Integer)
```

Reply:
Thanks, that's a good point. Also, we should also test that the two `Value` types are functionally equivalent. I decided not to focus on this in this PR in order to avoid it getting too large. Once I have a data-backed version of the `ScriptContext`, I will be able to see the bigger picture and start refactoring things to reduce code duplication and improve our tests. I'll add your suggestion to the list of things to do once I have this.

Reply:
> I suggest you to rewrite this test not to use golden file but to directly compare with constant

Normally I'd agree, but given that here we can instead check this same result in two different golden tests, I'd suggest we do that instead. But it's fine either way.